### ISSUE TITLE
ActionManager memory leak

### ISSUE THREAD

**Issue by SongCF** [cocos2d/cocos2d-x#14050]
when I create a node with an action, and not add to any other node, such as:

```
var sp = cc.Sprite.create("text.png");
sp.runAction(cc.fadeOut(10));
//has not parent
```

This will lead to memory leak, in correctly, 'sp' will be released in next frame, but when it run an action, 'sp' will not be released forever. 

//CCActionManager.cpp/addAction

```
void ActionManager::addAction(Action *action, Node *target, bool paused)
{
    //...
    HASH_FIND_PTR(_targets, &tmp, element);
    if (! element)
    {
        element = (tHashElement*)calloc(sizeof(*element), 1);
        element->paused = paused;

 // this code retain the 'sp', but released when action remove. if 'sp' has not parent, it not be running, so it's action has not been run
        target->retain();
        element->target = target;
        HASH_ADD_PTR(_targets, target, element);
    }
```

we solved by this code:

```
void ActionManager::update(float dt)
{
        // ...
        if (_currentTargetSalvaged && _currentTarget->actions->num == 0)
        {
            deleteHashElement(_currentTarget);
       }
//in next frame, if 'sp' refrence count == 1, remove it
//if some node refrence 'sp', it's refrence count >= 2
       else if (_currentTarget->target->getReferenceCount() == 1)
       {
        deleteHashElement(_currentTarget);
       }
}
```

**Comment by zilongshanren** [comment]
@SongCF 
Thanks, could you open a PR?

**Comment by ricardoquesada** [comment]
closed, since PR #14090 was already merged
